recipe/meta.yaml

```diff
@@ -21,6 +21,9 @@ source:
       # This patch can be removed in 10.4.0 but helps with compatibility for nccl
       # https://github.com/NVIDIA/nccl/issues/494#issuecomment-881074063
       - patches/PR100102-2.a1b3484a8e6c53c8084723e3f1738d402374198e.patch      # [gcc_version == "10.3.0"]
+      - patches/old-allow-commands-in-main-specfile.patch                      # [gcc_version != "12.0.0"]
+        # FIXME this will need to be extended for >= 12.0.0 but not sure how to do with selectors
```

Comment:
I wonder if some thing like int(gcc_version.split(".")) < 12 will work? Selectors are python expressions. This won't catch dev versions but it should be fine otherwise.

Reply:
I'll give that a try.  Thanks.

Reply:
I tried this:
```
-      - patches/old-allow-commands-in-main-specfile.patch                      # [gcc_version != "12.0.0"]
-        # FIXME this will need to be extended for >= 12.0.0 but not sure how to do with selectors
-      - patches/new-allow-commands-in-main-specfile.patch                      # [gcc_version == "12.0.0"]
+      - patches/old-allow-commands-in-main-specfile.patch                      # [int(gcc_version.split(".")[0]) <  int(12)]
+      - patches/new-allow-commands-in-main-specfile.patch                      # [int(gcc_version.split(".")[0]) >= int(12)]
```

And both selectors evaluate to `True` somehow.  I also noticed notes in the conda-build docs saying that Jinja gets applied before the selectors and I tried some other things using jinja and got both of them to not be applied.  If I define a gcc_mv in the conda_build_config.yaml and zip it into the tuple with gcc_version, I can get that to work correctly.  Since it only makes the variant names longer but hopefully doesn't impact anything else, I'm going to do that.